src/clothing_mod.cpp

```diff
@@ -0,0 +1,94 @@
+#include "clothing_mod.h"
+
+#include <string>
+#include <set>
+
+#include "generic_factory.h"
+#include "item.h"
+
+namespace
+{
+
+generic_factory<clothing_mod> all_clothing_mods( "clothing mods" );
+
+} // namespace
+
+/** @relates string_id */
+template<>
+bool string_id<clothing_mod>::is_valid() const
+{
+    return all_clothing_mods.is_valid( *this );
+}
+
+/** @relates string_id */
+template<>
+const clothing_mod &string_id<clothing_mod>::obj() const
+{
+    return all_clothing_mods.obj( *this );
+}
+
+void clothing_mod::load( JsonObject &jo, const std::string & )
+{
+    mandatory( jo, was_loaded, "flag", flag );
+    mandatory( jo, was_loaded, "item", item_string );
+    mandatory( jo, was_loaded, "implement_prompt", implement_prompt );
+    mandatory( jo, was_loaded, "destroy_prompt", destroy_prompt );
+
+    JsonArray jarr = jo.get_array( "mod_value" );
+    while( jarr.has_more() ) {
+        JsonObject mv_jo = jarr.next_object();
+        mod_value mv;
+        mandatory( mv_jo, was_loaded, "type", mv.type );
+        mandatory( mv_jo, was_loaded, "value", mv.value );
+        JsonArray jarr_prop = mv_jo.get_array( "proportion" );
+        while( jarr_prop.has_more() ) {
+            std::string str = jarr_prop.next_string();
+            if( str == "thickness" ) {
+                mv.thickness_propotion = true;
+            } else if( str == "coverage" ) {
+                mv.coverage_propotion = true;
+            }
```

Comment:
```suggestion
            } else {
                jarr_prop.throw_error( "Invalid value, valid are: \"coverage\" and \"thickness\"");
            }
```
So the user gets an information why their setting was ignored.